root_orchestrator/resource-abstractor/db/clusters_db.py

```diff
@@ -1,26 +1,95 @@
+from datetime import datetime
+
 import db.mongodb_client as db
 from bson.objectid import ObjectId
+from db.clusters_helper import get_freshness_threshold
+
+HISTORY_SLICE_SIZE = -100
+
+
+def create_cluster(data):
+    inserted = db.mongo_clusters.insert_one(data)
+
+    return db.mongo_clusters.find_one({"_id": inserted.inserted_id})
 
-from .clusters_helper import get_freshness_threshhold
+
+def update_cluster(cluster_id, data):
+    return db.mongo_clusters.find_one_and_update(
+        {"_id": ObjectId(cluster_id)},
+        {"$set": data},
+        return_document=True,
+    )
 
 
 def find_clusters(filter):
     pipeline = [
         {"$match": filter},
         {
             "$addFields": {
-                "active": {"$gt": ["$last_modified_timestamp", get_freshness_threshhold()]}
+                "active": {"$gt": ["$last_modified_timestamp", get_freshness_threshold()]}
             }
         },
     ]
-    return db.mongo_clusters.db.clusters.aggregate(pipeline)
+    return db.mongo_clusters.aggregate(pipeline)
 
 
 def find_cluster_by_id(cluster_id):
     cluster = list(find_clusters({"_id": ObjectId(cluster_id)}))
     return cluster[0] if cluster else None
 
 
-def find_cluster_by_name(name):
-    cluster = list(find_clusters({"cluster_name": name}))
-    return cluster[0] if cluster else None
+def find_cluster_by_name(cluster_name):
+    return db.mongo_clusters.find_one({"cluster_name": cluster_name})
+
+
+def get_update_data(data):
+    # Define a mapping from data keys to database keys
+    datetime_now = datetime.now()
+
+    key_mapping = {
+        "cpu_percent": "aggregated_cpu_percent",
+        "cpu_cores": "total_cpu_cores",
+        "memory_percent": "aggregated_memory_percent",
+        "cumulative_memory_in_mb": "memory_in_mb",
+        "number_of_nodes": "active_nodes",
+        "gpu_cores": "total_gpu_cores",
+        "gpu_percent": "total_gpu_percent",
+        "virtualization": "virtualization",
+        "more": "more",
+        "worker_groups": "worker_groups",
+        "aggregation_per_architecture": "aggregation_per_architecture",
+    }
+
+    # Use the mapping to create the update dictionary
+    update_dict = {db_key: data.get(data_key) for data_key, db_key in key_mapping.items()}
```

Comment:
doesn't it trigger an exception if one of the keys is missing from the update data?

Reply:
this is already the current implementation on `develop`: check out `cluster_db.py::mongo_update_cluster_information`. 
All I did was move this function somewhere else while renaming it and turning it into a map. Happy to update the logic if there is one.